src/ShoppingList.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
+import ListItems from './ListItems';
+
+class ShoppingList extends Component {
+  constructor(props) {
+    super(props);
+    this.addItem = this.addItem.bind(this);
+    this.state = { items: [] };
+  }
+
+  addItem(e) {
+    if (this._inputElement.value !== "") {
+      var newItem = {
```

Comment:
A small thing, you probably want to use `const` here if you don't plan to reassign the variable. It gives the variable 'block scope' (i.e. it can't use them outside this if statement) which is a lot more predictable and intuitive. 

```suggestion
      const newItem = {
```